lnwallet/commitment.go

```diff
@@ -650,13 +650,29 @@ func (c *CommitAuxLeaves) ForRemoteCommit() CommitAuxLeaves {
 //
 // TODO(roasbeef): move into pkg for custom chans?
 type AuxLeafStore interface {
-	// FetchLeaves attempts to fetch the auxiliary leaves for the
+	// FetchLeavesFromView attempts to fetch the auxiliary leaves that
+	// correspond to the passed aux blob, and pending fully evaluated HTLC
+	// view.
+	FetchLeavesFromView(prevBlob tlv.Blob,
+		view *HtlcView,
+		keyRing CommitmentKeyRing) fn.Option[CommitAuxLeaves]
+
+	// FetchLeavesFromCommit attempts to fetch the auxiliary leaves that
+	// correspond to the passed aux blob, and an existing channel
 	// commitment.
-	FetchLeaves(chanPoint wire.OutPoint,
-		auxBlob tlv.Blob) fn.Option[CommitAuxLeaves]
-
-	// RefreshBlob is used to refresh the blob with the new commitment.
-	RefreshBlob(newCommitTx *wire.MsgTx, oldBlob tlv.Blob) tlv.Blob
+	FetchLeavesFromCommit(c channeldb.ChannelCommitment,
+		keyRing CommitmentKeyRing) fn.Option[CommitAuxLeaves]
+
+	// FetchLeavesFromRevocation attempts to fetch the auxiliary leaves
+	// from a channel revocation that stores balance + blob information.
+	FetchLeavesFromRevocation(r *channeldb.RevocationLog,
+		keyRing CommitmentKeyRing) fn.Option[CommitAuxLeaves]
+
+	// ApplyHtlcView serves as the state transition function for the custom
+	// channel's blob. Given the old blob, and an HTLC view, then a new
+	// blob should be returned that reflects the pending updates.
+	ApplyHtlcView(oldBlob tlv.Blob, view *HtlcView,
```

Comment:
Where do we store the custom data (asset IDs, amounts and incomplete transition proofs) for each individual HTLC? Is the idea that we store that in the commitment's blob? Or within each `HtlcEntry`? Depending on the answer, we might need to change the signature of this method slightly. Otherwise, how would we be able to update/create the custom blob for each `HtlcEntry`?
Or maybe I'm misunderstanding something and the custom blob within `HtlcEntry` is just what was sent over the wire with the onion and doesn't need to be created/updated by the `AuxLeafStore`.